alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs

```diff
@@ -19,22 +19,21 @@
 
 module Cardano.Ledger.Alonzo.TxBody
   ( TxOut (TxOut, TxOutCompact),
-    TxBody
-      ( TxBody,
-        txinputs,
-        txinputs_fee,
-        txouts,
-        txcerts,
-        txwdrls,
-        txfee,
-        txvldt,
-        txUpdates,
-        txADhash,
-        mint,
-        exunits,
-        sdHash,
-        scriptHash
-      ),
+    TxBody (TxBody),
+    txinputs,
+    txinputs_fee,
+    txouts,
+    txcerts,
+    txwdrls,
+    txfee,
+    txvldt,
+    txUpdates,
+    txADhash,
+    txmint,
+    txExunits,
+    txsdHash,
+    txscriptHash,
```

Comment:
Having explicit projection functions defined in the TxBody pattern, is problematic since they become over constrained. They can be defined with no contraints at all, if we do them as separate functions.